edit_course/views.py

```diff
@@ -111,6 +112,7 @@ def form_valid(self, form):
                 nn = self.renumber_recursion(module, n)
                 if module['status'] != 'hidden':
                     n = nn
+        self.number_learning_objects_consecutively()
```

Comment:
Is this always required? It doesn't seem like the order can change unless either of the renumber buttons are pressed.

Reply:
The ordering can also change if the user manually edits the order of the learning objects in the Edit course view.

Reply:
I don't see how one edits the ordering through EditContentView. AFAIK that would happen through EditModelView. Though, now that I think about this a bit more, I don't think this logic should be in any of the views. If the ordering is added to the database, the numbering would need to be updated whenever the course content is updated, and that could technically happen from anywhere in the code. The numbering logic would then have to be in the model and manager classes. This complexity is probably the reason the existing order attribute doesn't actually contain the absolute order.

Reply:
That's true, yes, the actual changes are made in the EditModelView (I misleadingly called the page in the UI a view), I unfortunately didn't think very far about where the renumbering should be calculated; rather just thought that if it's done whenever the Edit course page is saved (or the renumbering buttons are pressed), that's probably good enough, typically the changes are imported anyway by rebuilding the course. And good point about the course content being possible to happen from nearly anywhere in the code. and probably the numbering logic should be done somewhere else.